node/container.go

```diff
@@ -336,9 +350,39 @@ func (a *HostAgent) PauseContainer(serviceID string, instanceID int) error {
 
 	// pause the running container
 	a.setInstanceState(serviceID, instanceID, service.StatePausing)
-	if err := attachAndRun(ctrName, svc.Snapshot.Pause); err != nil {
-		logger.WithError(err).Debug("Could not pause running container")
-		return err
+
+	go func() {
+		cmd := ""
+		if svc.Snapshot.Pause != "" {
+			cmd = svc.Snapshot.Pause + fmt.Sprintf(" & echo $! > %s", pidFile)
+		}
+		output, err := attachAndRun(ctrName, cmd)
+		done <- out{string(output), err}
+	}()
+
+	select {
+		case result := <- done: {
+			if result.err != nil {
+				logger.WithError(result.err).Warn("Could not run command: %s; %s", svc.Snapshot.Pause, ctrName)
+				// block to trigger timeout in facade/service.go
+				select{}
```

Comment:
Don't block, just return the error.  As long as we never call setInstanceState(...., service.StatePaused), the master will timeout waiting (which is what we want).  If you block forever here, we'll leak a goroutine.

Reply:
Also, we should try the unpause script as well here, and return the state to Running as you do below.